feat(gitlab): Use thread instead of comment for blocking comment (#15971)
renovatebot/renovate: docs/usage/configuration-options.md

Alias values must be properly formatted URIs.

> ## artifactUpdateWarningAsThread

Reviewer: I think we should prefix it with `gitlab` if there's no other platforms supporting it.

Is this for issues, PRs, or both?

Does it only apply to artifacts comments or others too?

> ## artifactUpdateWarningAsThread

Author: I'm fine with using the gitlab prefix. If other platforms support it in the future it might have to be renamed, but since it seems that only Gitlab supports this ATM it makes more sense to prefix it.

I'm not export on Gilab, or the GItlab API or terms, so I might be confused here sometimes; but AFAIK there is no "threads API". A _thread_ is a special type of discussion that contains one or more notes that can be marked as "resolved". A note then is a single reaction in a discussion. See https://docs.gitlab.com/ee/api/discussions.html (and challenge me if I'm wrong).
While a discussion may also be attached to issues (and epics, snippets and commits), I think that is irrelevant in this case.

Using the term "discussion" in this case is confusing, since a discussion is already how it is done. Also, in the UI for end-users the term "Thread" is used with the following explaination: "Discuss a specific suggesion or question that needs to be resolved." thus to me it makes sense to also use this term.

So in my opinion the name for the option should then be: `gitlabArtifactUpdateWarningAsThread`. Do you agree?

> By default, Renovate will create a warning comment when an artifact update occurs.

Reviewer: ```suggestion
By default, Renovate will create a warning comment when an artifact update fails.
```

> By configuring this setting to `true`, Renovatebot will instead create a thread when an artifact update occurs. This requires explicit acknowlegement of the issue when your repository is configured to have all threads resolved before merging.

Reviewer: ```suggestion
By configuring this setting to `true`, Renovate will instead create a thread when such an artifact update warning occurs.
```
